### ISSUE TITLE
[HASKELL] Update to newer servant version

### ISSUE THREAD

**Issue by loewenstein** [swagger-api/swagger-codegen#4162]
##### Description

Looking at taken from https://www.stackage.org/package/servant

```
LTS Haskell 7.8:	0.8.1
Stackage Nightly 2016-11-09:	0.9.1.1
Latest on Hackage:	0.9.1.1
```

shows we are one behind stack LTS and even two behind Hackage.

##### Swagger-codegen version

master

**Comment by wing328** [comment]
@loewenstein may I know if you have time to contribute the enhancement? 

Here is a good starting point: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/haskell-servant/stack.mustache

**Comment by wing328** [comment]
cc @algas

**Comment by wing328** [comment]
Closed via #5061 

Thanks for the PR from @mandrean
